arch/arm/core/aarch32/cortex_m/mpu/arm_core_mpu.c

```diff
@@ -148,6 +148,20 @@ void z_arm_configure_dynamic_mpu_regions(struct k_thread *thread)
 	 * of the respective dynamic MPU regions to be programmed for
 	 * the given thread. The array of partitions (along with its
 	 * actual size) will be supplied to the underlying MPU driver.
+	 *
+	 * The drivers of what regions get configured are CONFIG_USERSPACE,
+	 * CONFIG_MPU_STACK_GUARD, and K_USER/supervisor threads.
+	 *
+	 * If CONFIG_USERSPACE then any Memory Domains associated with the
+	 * thread get a defined region.
+	 *
+	 * If CONFIG_USERSPACE and the thread is a K_USER thread the usermode
```

Comment:
```suggestion
	 * If CONFIG_USERSPACE and the thread is a user thread, then a region is allocated for the user thread stack.
```

Reply:
Ok. But I kind of liked linking this to K_USER that is used in the thread creation